### ISSUE TITLE
MacDown should be more relaxed with LineBreak or EmptyLine between markup sequences: Bodytext, list, heading

### ISSUE THREAD

**Issue by porg** [MacDownApp/macdown#1277]
## Sample 
```
My list:
- Item one
- Item two should have this help text:
## Help text heading
Some prose here.
```

## Syntax Issue 1
`item one` requires empty line above to be recognized as bullet list. Already reported in #531

## Syntax Issue 2
`## Help text heading` requires an empty line above to be recognized as a headline.

## Side by side comparison of [QLMarkdown](https://github.com/sbarex/QLMarkdown) vs [MacDown](https://github.com/MacDownApp/macdown)

<img width="1500" alt="List LineBreak or EmptyLine Heading - MacDown very strict vs QLMarkdown with more relaxed interpretation" src="https://user-images.githubusercontent.com/737143/141798763-170202ec-fc0e-4365-892a-2e33bf45331f.png">

## Proposal
- Would be very desireable if the Mac QuickLook plugin QLMarkdown and the Mac Markdown WYSIWYG Editor MacDown would come to the same results.

## Considerations
- As far as I see it there's no problem with being more laissez fair with markdown interpretation:
  - If less (linebreak only) leads to the same result as more (empty line)
  - then adherents of both flavors could continue using their files and preferred markup and still get the same output (rendering).
  - Of course in a collaborative environment (such as source code management) they will get collisions, but for individual use of popular Mac read-only and edit+preview apps they will get consistent results at least, no need to enforce one style only.

**Comment by mmaunder** [comment]
Yeah this is making macdown unusable for me because LLMs generate the md files I read, and the issue where you need an additional linebreak (two linebreaks or newlines) after a heading for code blocks and bullets to render properly is making macdown very hard to use. I'm having to manually edit the md files generated. Sorry but "fix your LLM" is not appropriate because it's not fixable on the LLM side because they've been trained on markdown where not having an extra line break is so ubiquitous.
